pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -125,6 +124,12 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.TopologyManagerPolicy != kubeletconfig.NoneTopologyManagerPolicy && !localFeatureGate.Enabled(features.TopologyManager) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: TopologyManager %v requires feature gate TopologyManager", kc.TopologyManagerPolicy))
 	}
+	if localFeatureGate.Enabled(features.GracefulNodeShutdown) && kc.ShutdownGracePeriodCriticalPods.Duration > kc.ShutdownGracePeriod.Duration {
+		allErrors = append(allErrors, fmt.Errorf("invalid configuration: ShutdownGracePeriodCriticalPods (%v) must be <= (%v) ShutdownGracePeriod", kc.ShutdownGracePeriodCriticalPods, kc.ShutdownGracePeriod))
```

Comment:
Should we enforce a non-zero or a minimum for the critical pods?

Reply:
I added additional validation that `kc.ShutdownGracePeriod.Duration` and `kc.ShutdownGracePeriodCriticalPods.Duration` must be both be >= 0. 

Regarding enforcing non-zero or minimum for critical pods, I think >= 0 makes sense, for example some users might want to explicitly allocate all of the shutdown time for non-critical pods, so they could set `ShutdownGracePeriodCriticalPods=0`. WDYT?

Reply:
Sounds good!